core/src/main/scala/org/apache/spark/memory/StaticMemoryManager.scala

```diff
@@ -119,6 +119,21 @@ private[spark] object StaticMemoryManager {
    */
   private def getMaxExecutionMemory(conf: SparkConf): Long = {
     val systemMaxMemory = conf.getLong("spark.testing.memory", Runtime.getRuntime.maxMemory)
+    val minSystemMemory = (MemoryManager.RESERVED_SYSTEM_MEMORY_BYTES * 1.5).ceil.toLong
+
+    if (systemMaxMemory < minSystemMemory) {
```

Comment:
Hm maybe we could factor out these duplicated checks into the new `object`? You can use `require` to check a condition and if it fails throw IAE if it fails.